src/arrayMethodJoin.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const length = this.length;
+
+    let separUsed = separator;
+
+    if ('' + separUsed === 'undefined') {
+      separUsed = ',';
+    }
+
+    for (let i = 0; i < length; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
+        ;
+        // if equal to just pass. I could not make !equal to work
```

Comment:
typeof is really not needed here
also you can use an implicit conditional - `undefined` and `null` are equivalent to `false` in conditionals
so `this[i] === 'undefined' || this[i] === null` === `!this[i]`
```suggestion
      if (!this[i]) {
        result += separator;
        // if equal to just pass. I could not make !equal to work
```